### DISCUSSION TITLE
Dialect event "do_connect" no longer works with 'initdb' on Airflow v2.4.2

### DISCUSSION THREAD

**Discussion by MandarBathe** [apache/airflow#27791] [Q&A] [answered]
Hi - I register to a  'do_connect' _dialect event_ of SQLAlchemy to inject password field just before meta-db connection. It has been working fine till Airflow version 2.3.4. However it is not working as expected since version 2.4.2 as I see the initdb function now uses SQL_ALCHEMY_CONN directly to create _session_ table. Also I use securecookie for session_backend (not database).

Is there any workaround so that Airflow always uses db connection after processing do_connect event instead of using SQL_ALCHEMY_CONN directly?

**Comment by potiuk** [comment, marked as answer]
This has never been supported and it worked by accident. You should se standard mechanism provided by Airlfow. Unless you spend your own time to implement your own niche case and provide PR to make it works, I do not think we will invest in it  - this is the first time we hear of this case and if you still want to use it, contributing back fix seems like the best contribution back you can give for the free software you get.

**Reply by MandarBathe** [reply]
Thanks @potiuk

I have explored all standard mechanisms provided by Airflow. They do not fir my case, hence I will invest to solve it and raise PR here.
